### PR TITLE
Clean include

### PR DESCRIPTION
**Pull request by Janrupf**
This PR should fix some issues with includes (mostly these fixes are just for keeping them sorted better).
~~This PR won't fix the mixed use of `#pragma once` and include guards, we'll have to discuss whats better later.~~

EDIT: Will change to `#pragma once` for everything

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -59,16 +59,16 @@ ForEachMacros:
   - foreach
   - Q_FOREACH
   - BOOST_FOREACH
-IncludeBlocks:   Preserve
+IncludeBlocks:   Regroup
```
Comment (cwfitzgerald) [inline .clang-format:62]: Nah, this needs to be preserve in order to separate out problematic headers.

File: include/nova_renderer/nova_renderer.hpp [hunk 2]
```diff
@@ -1,12 +1,8 @@
 /*!
  * \brief Main class for Nova. This class exists as a singleton so it's always available
```
Comment (cwfitzgerald) [inline include/nova_renderer/nova_renderer.hpp]: If you're going to keep this it should be on the nova-renderer class.

**Review by cwfitzgerald — CHANGES_REQUESTED** [review]

File: .clang-format (hunk 1 above)
Comment (Janrupf) [reply .clang-format:62]: Not sure about that, in my opinion we should try to not depend on include order at all. At the moment it works, are you sure that should be changed back?

Comment (cwfitzgerald) [reply .clang-format:62]: I agree we should try not to and none of the headers we write will be, but some really really dumb headers exist that are not self-sufficient.

File: src/loading/shaderpack/json_interop.hpp [hunk 3]
```diff
@@ -32,6 +26,4 @@ namespace nova::renderer {
     void from_json(const nlohmann::json& j, material_pass& pass);
 
     void from_json(const nlohmann::json& j, material_data& mat);
-} // namespace nova::renderer
-
-#endif // NOVA_RENDERER_JSON_INTEROP_HPP
+} // namespace nova::renderer
```
Comment (DethRaid) [inline src/loading/shaderpack/json_interop.hpp]: Missing trailing newline

File: src/loading/shaderpack/render_graph_builder.cpp [hunk 4]
```diff
@@ -1,11 +1,9 @@
-/*!
- * \author ddubois
- * \date 17-Sep-18.
- */
-
 #include "render_graph_builder.hpp"
```
Comment (DethRaid) [inline src/loading/shaderpack/render_graph_builder.cpp:1]: Why didn't this get sorted to the bottom of the includes?

File: src/loading/shaderpack/shaderpack_validator.hpp [hunk 5]
```diff
@@ -6,12 +6,12 @@
  * \date 12-Sep-18.
```
Comment (DethRaid) [inline src/loading/shaderpack/shaderpack_validator.hpp]: Should remove the author/date header comment

File: src/render_engine/vulkan/vulkan_render_engine.cpp [hunk 6]
```diff
@@ -1,25 +1,30 @@
...
 
+#include <spirv_cross/spirv_glsl.hpp>
+
+#include "nova_renderer/util/platform.hpp"
+
+#include "../../loading/shaderpack/render_graph_builder.hpp"
+#include "../../loading/shaderpack/shaderpack_loading.hpp"
+#include "../../util/logger.hpp"
+
+// TODO: Move windowing out of render engine folders to prevent things like this
```
Comment (DethRaid) [inline src/render_engine/vulkan/vulkan_render_engine.cpp]: Please define "things like this"

Comment (DethRaid) [reply src/render_engine/vulkan/vulkan_render_engine.cpp]: And say why they're bad

File: src/util/logger.hpp [hunk 7]
```diff
@@ -1,19 +1,14 @@
...
 
 #include <functional>
 #include <mutex>
 #include <sstream>
 #include <string>
 #include <unordered_map>
+#include <utility>
 
+// Hope this fixes windows build for now, not sure where it comes from
+// TODO: Fix the root of this issue (@dethraid, @cwfitzgerald)
```
Comment (DethRaid) [inline src/util/logger.hpp]: The root of the issue is that windows.h is stupid. We need to undefine ERROR

Comment (DethRaid) [reply src/util/logger.hpp]: Please change these comments to something like "windows.h defines ERROR, and Nova uses ERROR as a lot level, so we need to undefine ERROR so we can it"

**Review by DethRaid — CHANGES_REQUESTED** [review]

File: src/loading/shaderpack/render_graph_builder.cpp (hunk 4 above)
Comment (Janrupf) [reply src/loading/shaderpack/render_graph_builder.cpp:1]: Because its the header belonging to the source file

**Review by DethRaid — APPROVED** [review]

**Review by cwfitzgerald — APPROVED** [review]

**Pull request merged**